server/src/main/java/org/opensearch/action/search/SearchPhase.java

```diff
@@ -34,14 +34,15 @@
 import org.opensearch.common.CheckedRunnable;
 
 import java.io.IOException;
+import java.util.Locale;
 import java.util.Objects;
 
 /**
  * Base class for all individual search phases like collecting distributed frequencies, fetching documents, querying shards.
  *
  * @opensearch.internal
  */
-abstract class SearchPhase implements CheckedRunnable<IOException> {
+public abstract class SearchPhase implements CheckedRunnable<IOException> {
     private final String name;
```

Comment:
Do you think in the long run it would make sense to make this member a `SearchPhaseName`, since it's more type-safe than relying on a string? If so, maybe we should add a `TODO` to that effect.

Since this class hasn't been `public` (until now), we may be able to make that change now without breaking consumers. What do you think?

Reply:
Adding a TODO, yes we can do that. 

Using enum instead of String, yes it is far better in terms of Type checking etc. But should we do this in PR is what I am not sure. I would like to take that refactoring as part of next PR. The reason for that is because there are many places in tests, where we have used "test", as search phase name. We need to see what we want to do for those cases. Should we use an already defined name like Query, Fetch etc or should I create a SearchPhase name called as Test for unit tests and ITs.

I think there are some decisions to be made here. Hence I would like to consider that change as a separate PR.